pallets/stateful-storage/src/types.rs

```diff
@@ -19,41 +19,108 @@ pub struct ItemHeader {
 	payload_len: u16,
 }
 
+/// This header is used to specify how much data is in a page and is inserted into the buffer
+/// before every page in a Paginated model.
+#[derive(Clone, Encode, Decode, PartialEq, MaxEncodedLen, Debug)]
+pub struct PageHeader {
+	/// The length of this item, not including the size of this header.
+	payload_len: u16,
+}
+
 #[derive(Debug, PartialEq)]
-pub enum ItemPageError {
+pub enum PageError {
 	ErrorParsing(&'static str),
 	InvalidAction(&'static str),
 	ArithmeticOverflow,
+	PageSizeOverflow,
 }
 
-/// A page of items
+// NOTE: Would prefer to simply have an enum:
+// ```no_compile
+// enum PageType { Itemized, Paginated }
+// ```
+// However, currently can't use enum variants as const generic parameters.
+pub trait PageType {}
```

Comment:
I think this is a overkill for now. Let's implement a trait when there is a common functionality we need on both

Reply:
There is common functionality already:
* `From<BoundedVec>`
* `TryFrom<Vec>`

There is also functionality that differentiates on `PageType`:
* `parse()` (though I think this might do well as a `TryFrom` train on the `Parsed*Page` struct)